client/src/App.jsx

```diff
@@ -1,11 +1,54 @@
+import { useState, useEffect } from 'react';
 import './App.css'
+import Navbar from './components/Navbar'
+import ContactCard from './components/ContactCard';
 
 function App() {
 
+  const [contacts, setContacts] = useState([]);
+
+  async function getContacts() {
+    try {
+      const response = await fetch('http://localhost:1212');
+
+      if(!response.ok) {
+        throw new Error('response was not ok');
+      }
+
+      const allContacts = await response.json();
+      setContacts(allContacts)
+      console.log(contacts[0])
+
+    } catch (error) {
+      console.log(error.message);
+    }
+  }
+
+  // function update(editedContact) {
+  //   setContacts((prevContacts) => 
+  //     prevContacts.map((contact) => 
+  //       contact.id === editedContact.id ? editedContact : contact
+  //     )
+  //   );
+  // }
+  
+
+  useEffect(() => {
+    getContacts();
+  }, []);
+
   return (
-    <>
-      <p>Hello</p>
-    </>
+    <div>
+      <Navbar />
+      <div  className="cards-container">
+        {contacts.length > 0 ? 
+          contacts.map((item, index) => (
+            <ContactCard key={index} contacts = {item}  />
+          )) : ` `}
```

Comment:
No need to change anything, but calling `.map()` on an empty array will return an empty array, so there's no need to do the `contacts.length > 0` check in this case. You only want to do that check if instead of `` : ` ` ``, you want to display a message that says "No contacts"

Reply:
Ok, I'll remember this in the future! Thank you!